doc/zarafa.1
'\" t
.\"     Title: zarafa
.\"    Author: [see the "Author" section]
.\" Generator: DocBook XSL Stylesheets v1.76.1 <http://docbook.sf.net/>
.\"      Date: August 2011
.\"    Manual: Zarafa user reference
.\"    Source: Zarafa 7.1
.\"  Language: English
.\"
.TH "ZARAFA" "1" "August 2011" "Zarafa 7.1" "Zarafa user reference"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
zarafa \- Zarafa server overview\&.
.SH "DESCRIPTION"
.PP
Zarafa is a full\-featured e\-mail and groupware solution, focused towards clients using the MAPI standard\&. The Zarafa suite consists of the following parts:
.PP
zarafa\-server
.RS 4
The main server process, accepting client connections, and managing storage and retrieval of data from the database\&.
.RE
.PP
zarafa\-spooler
.RS 4
The outgoing e\-mail spooler\&. This service makes sure that e\-mails sent by clients are converted to internet e\-mail and forwarded to an SMTP server\&.
.RE
.PP
zarafa\-dagent
.RS 4
The Zarafa delivery agent is called by your MTA to deliver an incoming e\-mail message into a user\*(Aqs inbox\&.
.RE
.PP
zarafa\-admin
.RS 4
The admin program can be started from the command line to perform various administrational tasks such add creating users, and managing groups\&.
.RE
.PP
zarafa\-monitor
.RS 4
The monitor server process monitors user stores for quota limits\&. When a user goes over the warning quota level, zarafa\-monitor will issue a warning e\-mail\&.
.RE
.PP
zarafa\-gateway
.RS 4
The gateway server process enables POP3 and/or IMAP connectivity to the Zarafa server\&. Users with other non\-Outlook or webaccess clients can still retreive their e\-mail from the server\&.
.RE
.PP
zarafa\-passwd
.RS 4
The passwd program can be started from the command line to update the user password when the server is running with dbplugin\&.
.RE
.PP
zarafa\-fsck
.RS 4
The fsck program can be started from the command line to check and possibly fix broken items in your calendar or tasks folder\&.
.RE
.PP
zarafa\-backup
.RS 4
The backup tool creates an incremental dump of a store\&. The data can be restored by the
\fBzarafa\-restore\fR
tool\&. This tool can be used as a brick\-level\-backup\&. It can not be used as a disaster\-recovery tool\&.
.RE
.PP
zarafa\-restore
.RS 4
The restore tool reads data from a backup create by
\fBzarafa\-backup\fR\&. You can restore single items or folders\&. To restore a complete store, you can use the example script found in
/usr/share/zarafa/zarafa\-backup\-helpers\&.
.RE
.PP
zarafa\-ical
.RS 4
The iCal/CalDAV gateway enables users to retrieve their own Zarafa calendar using the iCalendar standard\&.
.RE
.PP
zarafa\-licensed
.RS 4
The zarafa\-licensed daemon is used for licensing query information from non\-free parts of the zarafa product\&. You must run this daemon if you wish to use non\-free components of the zarafa product, like access from Windows environments\&.
.RE
.PP
zarafa\-report
.RS 4
A reporting program used by zarafa\-licensed to send license reports to the zarafa\-server using a secured connection\&.
.RE
.PP
zarafa\-stats
.RS 4
A simple program that dumps some statistics the zarafa\-server program collects\&.
.RE
.PP
zarafa\-cfgchecker
.RS 4
This program tests for common configuration errors for a given config file\&.`
.RE
.PP
zarafa\-search
.RS 4
The zarafa\-search daemon is used to index all messages for all users in the zarafa\-server\&. Indexing messages greatly enhances the search performance of the zarafa\-server\&.
.RE
.PP
zarafa\-ws
.RS 4
The Zarafa Web Services server process enables EWS connectivity to the Zarafa server\&. Users with other non\-Windows Outlook or WebApp clients can still retreive their e\-mail and calendars from the server\&.
.RE
.SH "AUTHOR"
.PP
Written by Zarafa\&.
.SH "SEE ALSO"
.PP

\fBzarafa-server\fR(1)
\fBzarafa-spooler\fR(1)
\fBzarafa-dagent\fR(1)
\fBzarafa-admin\fR(1)
\fBzarafa-monitor\fR(1)
\fBzarafa-gateway\fR(1)
\fBzarafa-passwd\fR(1)
\fBzarafa-fsck\fR(1)
\fBzarafa-backup\fR(1)
\fBzarafa-restore\fR(1)
\fBzarafa-ical\fR(1)
\fBzarafa-licensed\fR(1)
\fBzarafa-report\fR(1)
\fBzarafa-stats\fR(1)
\fBzarafa-cfgchecker\fR(1)
\fBzarafa-search\fR(1)
\fBzarafa-ws\fR(1)


debian/changelog
giraffe (7.1.5.final-1~1.gbpdcf72d) UNRELEASED; urgency=low

  ** SNAPSHOT build @dcf72d11495083f2275fa118afad08bba3903cfb **

  [ Guido Günther ]
  * [6dbdff8] Change project name to giraffe
  * [ce5202f] Change maintainer to pkg-giraffe
  * [b691d5e] Switch DH_VERBOSE to false
  * [11a4382] Don't try to remove non existant directory

  [ Matthias Geerdsen ]
  * [2a1aa8f] fix libboost dependencies for sid
  * [7dd25fa] zarafa-webaccess-mobile is no longer shipped
  * [a90de06] use autotools-dev debhelpers for config.{sub,guess} updating
  * [d09323c] fix installation of libzarafasync
  * [072134d] rebase patch to new upstream
  * [11d8d22] add patch to fix casting problem with gzfile (from Mandriva)
  * [0f96759] add patch to fix php compatibilty issue pval -> zval
  * [ee82ed2] fix libvime0 for version in sid, add zarafa-contacts package
  * [6a5a9fb] add zarafa-contacts package
  * [c6d60f3] add /etc/mapi to zarafa-client
  * [e73e75a] fix typo in package name
  * [05eaf43] add patch to remove hex2bin() declaration, it is available in php 5.4.0

  [ Robert Kehl ]
  * Renaming zarafa to giraffe
  * Renaming zarafa-common to giraffe-common
  * Renaming zarafa-server to giraffe-server
  * Renaming zarafa-dbg to giraffe-dbg
  * Renaming zarafa-dev to giraffe-dev
  * Renaming zarafa-client to giraffe-client
  * Renaming zarafa-contacts to giraffe-contacts
  * Renaming zarafa-libs to giraffe-libs
  * Renaming zarafa-libarchiver to giraffe-libarchiver
  * Renaming zarafa-utils to giraffe-utils
  * Renaming zarafa-monitor to giraffe-monitor
  * Renaming zarafa-spooler to giraffe-spooler
  * Renaming zarafa-dagent to giraffe-dagent
  * Renaming zarafa-gateway to giraffe-gateway
  * Renaming zarafa-ical to giraffe-ical
  * Renaming zarafa to giraffe
  * Renaming zarafa-archiver to giraffe-archiver
  * Renaming zarafa-webaccess to giraffe-webaccess
  * Renaming zarafa-search to giraffe-search
  * Clarifying package descriptions
  * fixing file names in conf files
  * fix rename

 -- Robert Kehl <robertkehl@robertkehl.de>  Sat, 29 Mar 2014 14:39:50 +0100

giraffe (7.0.3~beta1-1~1.gbpe4c14c) UNRELEASED; urgency=low

  ** SNAPSHOT build @e4c14c384782481ae8adde0c376927d45caca787 **

  * [e4c14c3] Add gbp.conf

 -- Guido Günther <agx@sigxcpu.org>  Thu, 17 Nov 2011 20:30:30 +0100

zarafa (7.0.1-28228) unstable; urgency=low

  * new release, build using libgoogle-perftools

 -- John van der Kamp <john@memleak.office.zarafa.com>  Wed, 20 Jul 2011 16:20:09 +0200

zarafa (7.0.0-27791.1) unstable; urgency=low

  * Non-maintainer upload.
  * Make 3.0 (quilt) source format

 -- John van der Kamp <john@memleak.office.zarafa.com>  Fri, 15 Jul 2011 16:36:34 +0200

zarafa (7.0.0-27791) unstable; urgency=low

  * First OpenSource release of Zarafa Collaboration Platform for Debian.

 -- John van der Kamp <j.vdkamp@zarafa.com>  Fri, 15 Jul 2011 15:28:10 +0200


php-webclient-ajax/zarafa.php
<?php
/*
 * Copyright 2005 - 2014  Zarafa B.V.
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License, version 3, 
 * as published by the Free Software Foundation with the following additional 
 * term according to sec. 7:
 *  
 * According to sec. 7 of the GNU Affero General Public License, version
 * 3, the terms of the AGPL are supplemented with the following terms:
 * 
 * "Zarafa" is a registered trademark of Zarafa B.V. The licensing of
 * the Program under the AGPL does not imply a trademark license.
 * Therefore any rights, title and interest in our trademarks remain
 * entirely with us.
 * 
 * However, if you propagate an unmodified version of the Program you are
 * allowed to use the term "Zarafa" to indicate that you distribute the
 * Program. Furthermore you may use our trademarks where it is necessary
 * to indicate the intended purpose of a product or service provided you
 * use it in accordance with honest practices in industrial or commercial
 * matters.  If you want to propagate modified versions of the Program
 * under the name "Zarafa" or "Zarafa Server", you may only do so if you
 * have a written permission by Zarafa B.V. (to acquire a permission
 * please contact Zarafa at trademark@zarafa.com).
 * 
 * The interactive user interface of the software displays an attribution
 * notice containing the term "Zarafa" and/or the logo of Zarafa.
 * Interactive user interfaces of unmodified and modified versions must
 * display Appropriate Legal Notices according to sec. 5 of the GNU
 * Affero General Public License, version 3, when you propagate
 * unmodified or modified versions of the Program. In accordance with
 * sec. 7 b) of the GNU Affero General Public License, version 3, these
 * Appropriate Legal Notices must retain the logo of Zarafa or display
 * the words "Initial Development by Zarafa" if the display of the logo
 * is not reasonably feasible for technical reasons. The use of the logo
 * of Zarafa in Legal Notices is allowed for unmodified and modified
 * versions of the software.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *  
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 * 
 */

?>
<?php
	/**
	* This file is the dispatcher of the whole application, every request for data enters 
	* here. XML is received and send to the client.
	*/
	
	// Include files
	include("config.php");
	include("defaults.php");
	include("server/util.php");
	require("server/PEAR/JSON.php");
	
	require("mapi/mapi.util.php");
	require("mapi/mapicode.php");
	require("mapi/mapidefs.php");
	require("mapi/mapitags.php");
	require("mapi/mapiguid.php");

	require("server/core/class.conversion.php");
	require("server/core/class.mapisession.php");
	require("server/core/class.entryid.php");
	
	include("server/core/constants.php");
	
	include("server/core/class.state.php");
	include("server/core/class.attachmentstate.php");
	include("server/core/class.request.php");
	include("server/modules/class.module.php");
	include("server/modules/class.listmodule.php");
	include("server/modules/class.itemmodule.php");
	include("server/core/class.operations.php");
	include("server/core/class.properties.php");
	include("server/core/class.tablecolumns.php");
	include("server/core/class.bus.php");
	include("server/core/class.settings.php");
	include("server/core/class.language.php");
	include("server/core/class.pluginmanager.php");
	include("server/core/class.plugin.php");

	ob_start();
	setlocale(LC_CTYPE, "en_US.UTF-8");

	// Set timezone
	if(function_exists("date_default_timezone_set")) {
		if(defined('TIMEZONE') && TIMEZONE) {
			date_default_timezone_set(TIMEZONE);
		} else if(!ini_get('date.timezone')) {
			date_default_timezone_set('Europe/London');
		}
	}

	// Get the available modules
	$GLOBALS["availableModules"] = getAvailableModules();

	// Callback function for unserialize
	// Module objects of the previous request are stored in the session. With this
	// function they are restored to PHP objects.
	ini_set("unserialize_callback_func", "sessionModuleLoader");
	
	// Start session
	session_name(COOKIE_NAME);
	session_start();
	
	// Create global mapi object. This object is used in many other files
	$GLOBALS["mapisession"] = new MAPISession();
	// Logon, the username and password are set in the "index.php" file. So whenever
	// an user enters this file, the username and password whould be set in the $_SESSION
	// variable
	if (isset($_SESSION["username"]) && isset($_SESSION["password"])) {
		$sslcert_file = defined('SSLCERT_FILE') ? SSLCERT_FILE : null;
		$sslcert_pass = defined('SSLCERT_PASS') ? SSLCERT_PASS : null;
		$hresult = $GLOBALS["mapisession"]->logon($_SESSION["username"], $_SESSION["password"], DEFAULT_SERVER, $sslcert_file, $sslcert_pass);
	}else{
		$hresult = MAPI_E_UNCONFIGURED;
	}

	if(isset($_SESSION["lang"])) {
		$session_lang = $_SESSION["lang"];
	}else{
		$session_lang = LANG;
	}
	
	// Close the session now, so we're not blocking other clients
	session_write_close();

	// Set headers for XML
	header("Content-Type: text/xml; charset=utf-8");
	header("Expires: ".gmdate( "D, d M Y H:i:s")."GMT");
	header("Last-Modified: ".gmdate( "D, d M Y H:i:s")."GMT");
	header("Cache-Control: no-cache, must-revalidate");
	header("Pragma: no-cache");
	header("X-Zarafa: ".phpversion("mapi").(defined("SVN") ? "-".SVN:""));
		
	// Check is the user is authenticated
	if ($GLOBALS["mapisession"]->isLoggedOn()) {
		// Authenticated
		// Execute request
		
		// Instantiate Plugin Manager
		$GLOBALS['PluginManager'] = new PluginManager();
		$GLOBALS['PluginManager']->detectPlugins();
		$GLOBALS['PluginManager']->initPlugins();
		// Get the available plugin modules
		$GLOBALS["availablePluginModules"] = $GLOBALS['PluginManager']->getAvailablePluginModules();

		// Create global operations object
		$GLOBALS["operations"] = new Operations();
		// Create global properties object
		$GLOBALS["properties"] = new Properties();
		// Create global tablecolumns object
		$GLOBALS["TableColumns"] = new TableColumns();
		// Create global settings object
		$GLOBALS["settings"] = new Settings();

		// Create global language object
		$GLOBALS["language"] = new Language();
		// Set the correct language
		$GLOBALS["language"]->setLanguage($session_lang);

		// Get the state information for this subsystem
		if(isset($_GET["subsystem"]))
			$subsystem = $_GET["subsystem"];
		else
			$subsystem = "anonymous"; // Currently should never happen	

		$state = new State($subsystem);
		
		// Lock the state of this subsystem
		$state->open();
		
		// Get the bus object for this subsystem
		$bus = $state->read("bus");

		if(!$bus)
			// Create global bus object
			$bus = new Bus();
		
		// Make bus global
		$GLOBALS["bus"] = $bus;
		
		// Reset any spurious information in the bus state
		$GLOBALS["bus"]->reset();
		
		// Create new request object
		$request = new Request();
		
		// Get the XML from the client
		$xml = readXML();
		if (function_exists("dump_xml")) dump_xml($xml,"in"); // debugging
		
		// Execute the request
		$xml = $request->execute($xml);

		/*
		 * If we get any errors with more unicode characters then also reject overly long
		 * 2 byte sequences, as well as characters above U+10000 and replace with ?
		
		$xml = preg_replace('/[\x00-\x08\x10\x0B\x0C\x0E-\x19\x7F]'.
		 '|[\x00-\x7F][\x80-\xBF]+'.
		 '|([\xC0\xC1]|[\xF0-\xFF])[\x80-\xBF]*'.
		 '|[\xC2-\xDF]((?![\x80-\xBF])|[\x80-\xBF]{2,})'.
		 '|[\xE0-\xEF](([\x80-\xBF](?![\x80-\xBF]))|(?![\x80-\xBF]{2})|[\x80-\xBF]{3,})/S',
		 '?', $xml );
		 */

		//reject overly long 3 byte sequences and UTF-16 surrogates and replace with ?
		$xml = preg_replace('/\xE0[\x80-\x9F][\x80-\xBF]'. // Remove strings like \xE0\x9F\xBF
		 '|\xEF[\xA0-\xBF][\x80-\xBF]'. // Remove strings like \xEF\xBF\xBF
		 '|\xED[\xA0-\xBF][\x80-\xBF]/S','?', $xml ); // Remove strings like \xED\xBF\xBF

		if (function_exists("dump_xml")) dump_xml($xml,"out"); // debugging

		// Check if we can use gzip compression
		if ((!defined("DEBUG_GZIP")||DEBUG_GZIP) && $GLOBALS["settings"]->get("global/use_gzip","true")=="true" && function_exists("gzencode") && isset($_SERVER["HTTP_ACCEPT_ENCODING"]) && strpos($_SERVER["HTTP_ACCEPT_ENCODING"], "gzip")!==false){
			// Set the correct header and compress the XML
			header("Content-Encoding: gzip");
			echo gzencode($xml);
		}else {
			echo $xml;
		}
		
		// Reset the BUS before saving to state information
		$GLOBALS["bus"]->reset();

		if(isset($GLOBALS["bus"]))
			$state->write("bus", $GLOBALS["bus"]);		

		// You can skip this as well because the lock is freed after the PHP script ends
		// anyway.
		$state->close();

	} else {
		echo "<zarafa>\n";
		echo "\t<error logon=\"false\" mapi=\"".get_mapi_error_name($hresult)."\">Logon failed</error>\n";
		echo "</zarafa>";
	}
?>
